table.go

```diff
@@ -69,17 +65,25 @@ func NumDigits(b *big.Int) int64 {
 	if bl == 0 {
 		return 1
 	}
-	if val, ok := lookupBits(bl); ok {
-		ab := new(big.Int).Abs(b)
-		if ab.Cmp(&val.border) < 0 {
+
+	if bl < digitsTableLen {
+		val := digitsLookupTable[bl]
+		a := b
+		if b.Sign() < 0 {
+			a = new(big.Int).Abs(b)
+		}
+		if a.Cmp(&val.border) < 0 {
 			return val.digits
 		}
 		return val.digits + 1
 	}
 
 	n := int64(float64(bl) / digitsToBitsRatio)
-	a := new(big.Int).Abs(b)
-	e := new(big.Int).Exp(bigTen, big.NewInt(n), nil)
+	e := tableExp10(n, nil)
+	a := b
+	if b.Sign() < 0 {
+		a = new(big.Int).Abs(b)
```

Comment:
If we're going to allocate `a`, why don't we use it as the tmp variable for `tableExp10` as well? 

Also, why does that function say `"The returned value must not be mutated"`? Seems unclear.